### ISSUE TITLE
Trouble if some stations contain station indexes, while others don't (?)

### ISSUE THREAD

**Issue by gwselke** [manwar/Map-Tube#15]
`map-tube --map=Singapore --generate_map`
reports the following:
Map has link from "Tanah Merah" to "Expo" but no lines in common

The map data for Singapore contain (among others) the following lines:

```
<station id="EW4" name="Tanah Merah" line="EWL:4" link="EW3,EW5,CG1"/> 
<station id="CG1" name="Expo" line="EWL" link="EW4,CG2"/>  
```
That is, both stations are in fact on line EWL, and Tanah Merah (id EW4) links indeed to Expo (id CG1) and the other way round.
The only unusual thing is that Tanah Merah also specifies a station index for its position on EWL, whereas Expo doesn't. 
This is probably an oversight by the map author, but since the index is optional, shouldn''t Map::Tube be able to cope withn this?
(Unless there is some other reason behind the warning from map-tube.)

**Comment by manwar** [comment]
@gwselke Good catch, the map data is not following the standards that is used by most (specially me). Having said, it is not documented so it is very easy ignore it.

The following two stations are the root cause and not following the standards (undocumented).

           <station id="CG1" name="Expo" line="EWL" link="EW4,CG2"/>
           <station id="CG2" name="Changi Airport" line="EWL" link="CG1"/>

The `line` attribute is missing station index. I would have expected it to be `line="EW:30"` and `line="EW:31"` respectively. Also I don't see why station id is not `EW30` and `EW31` either.

**Comment by gwselke** [comment]
This should be resolved in [Map::Tube::Singapore](https://metacpan.org/pod/Map::Tube::Singapore) (and possibly others).

**Comment by manwar** [comment]
@gwselke is it safe to close the issue now?

**Comment by gwselke** [comment]
Well, I guess it's a decision to be made between two positions regarding the rule "Either index all stations on a line, or none at all":

1. It is a documented, testable rule which the map maker **must** follow. If not, the map will be considered faulty and unusable. (Presumably, the map author will note at some time, and fix the issue.)
2. It is a documented rule which the map maker **should** follow. If not, however, Map::Tube will be forgiving and automatically fill in the gaps at map instantiation time, and no harm will be done. (Presumably, the map author will never bother to fix the issue, because it's workimg anyway.)

I think both positions have something going for them. If you go with 1), then, by all means, do close this issue!

Disclaimer: I do not understand the inner workings of M:T well enough to see whether the indexing (which, after all, is optional) adds some specific value to the workings of M::T, and whether filling in missing indexes is practically feasible.
